src/cloud-sql-instance.ts

```diff
@@ -106,17 +107,43 @@ export class CloudSQLInstance {
     }) as ReturnType<typeof pThrottle>;
   }
 
-  forceRefresh(): Promise<RefreshResult> {
+  forceRefresh() {
     // if a refresh is already ongoing, just await for its promise to fulfill
     // so that a new instance info is available before reconnecting
     if (this.next) {
-      return this.next;
+      return;
     }
     this.cancelRefresh();
-    return this.refresh();
+    this.scheduleRefresh(0);
+  }
+
+  // refreshComplete Returns a promise that resolves when the current refresh
+  // cycle has completed, either with success or failure. If no refresh is
+  // in progress, the promise will resolve immediately.
+  refreshComplete(): Promise<void> {
```

Comment:
I don't fully understand this... how is this different then awaiting the promise of `forceRefresh`?

Reply:
Good point. I've updated forceRefresh() to include all the logic. 

Basically, forceRefresh() only needs to return a promise for testing. forceRefresh is used as fire-and-forget in the main connector code.

In tests, the promise should resolve regardless of whether it succeeds or fails. Otherwise, it could cause tests cases to hang where refresh fails.